VisualizationBase/src/nodes/InfoNode.h

```diff
@@ -0,0 +1,116 @@
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+***********************************************************************************************************************/
+#pragma once
+
+#include "../VisualizationBase/src/visualizationbase_api.h"
+#include "ModelBase/src/nodes/Node.h"
+#include "ModelBase/src/nodes/TypedList.h"
+#include "ModelBase/src/nodes/nodeMacros.h"
+
+DECLARE_TYPED_LIST(VISUALIZATIONBASE_API, Visualization, InfoNode)
+
+namespace Visualization {
+
+/**
+ * An InfoNode is the base class of all nodes which display some sort of statistics
+ * on a target node. There is a register which keeps track of all possible informations
+ * to be displayed.
+ */
+class VISUALIZATIONBASE_API InfoNode : public Super<Model::Node>
+{
+	DECLARE_TYPE_ID
+
+	public:
+		typedef QString (*InfoGetter) (Model::Node* target);
+
+		InfoNode(Model::Node* target);
+		InfoNode(Model::Node *target, QList<QString> enabledInfos);
+		virtual ~InfoNode();
+
+		const Model::Node* target() const;
+
+		/**
+		 * If the statistics can be represented as a string, use
+		 * this to represent them as HTML text. Then the same visualization
+		 * can be used for multiple statistics nodes.
+		 */
+		QString infoHtml() const;
+		/**
+		 * Updates the node's stored information.
+		 * @param isAutoUpdate Is this an automatic update, or a forced update?
+		 *					   A forced update will update everything.
+		 */
+		virtual void updateInfo(bool isAutoUpdate);
+
+		void setEnabled(const QString name, bool isEnabled);
+		bool isEnabled(const QString name) const;
+
+		virtual void save(Model::PersistentStore& store) const;
+		virtual void load(Model::PersistentStore& store);
+
+		/**
+		 * Registers a new method to get information.
+		 * @param name The name of the getter.
+		 * @param getter The getter method itself, taking a node and returning a string.
+		 * @param updatesAutomatically Whether this info should always update.
+		 * @param enabledByDefault Whether this info is visible by default.
+		 */
+		static void registerInfoGetter(const QString& name, const InfoGetter getter,
+									   bool updatesAutomatically, bool enabledByDefault);
+
+	protected:
+		void setInfoHtml(QString content);
+
+	private:
+		QString infoHtml_;
+		QList<QString> enabledInfoGetters_;
+		QHash<QString, QString> cachedInfoStrings_;
+		Model::Node* target_{};
+
+		struct InfoGetterStruct {
+				InfoGetter getter_;
+				bool updatesAutomatically_{};
+				bool enabledByDefault_{};
+				InfoGetterStruct(InfoGetter getter, bool updatesAutomatically, bool enabledByDefault)
+						: getter_(getter), updatesAutomatically_(updatesAutomatically), enabledByDefault_(enabledByDefault)
+				{ }
+				InfoGetterStruct() {}
+		};
+		static QList<InfoNode*> allInfoNodes;
+		static QHash<QString, InfoGetterStruct> allInfoGetters;
+
+};
+
+inline void InfoNode::save(Model::PersistentStore &) const { Q_ASSERT(false); }
```

Comment:
Do not create inline virtual methods. Please put these in the .cpp file and do not inline them.